File: examples/access-levels/access-levels.ts
import { deploy, out } from '../../helper/typescript/typescript_test';
import { Statement } from '../../lib';

function getStatements() {
  function wrap() {
    // doc-start
    const s1 = new Statement.S3() //
      .deny()
      .allPermissionManagementActions();

    const s2 = new Statement.S3() //
      .allow()
      .allListActions()
      .allReadActions();
    // doc-end
    return [s1, s2];
  }
  return wrap();
}
const s = getStatements();
out(s);
deploy(s);

File: examples/access-levels/access-levels.py
import iam_floyd as statement
import importlib
import os
import sys
import inspect

currentdir = os.path.dirname(os.path.abspath(
    inspect.getfile(inspect.currentframe())))
helperDir = '%s/../../helper/python' % currentdir
sys.path.insert(0, helperDir)

test = importlib.import_module('python_test')
out = getattr(test, 'out')
deploy = getattr(test, 'deploy')


def statements():
    # doc-start
    s1 = statement.Ec2() \
                .deny() \
                .all_permission_management_actions()

    s2 = statement.Ec2() \
                .allow() \
                .all_list_actions() \
                .all_read_actions()
    # doc-end
    return [s1, s2]


all = statements()
out(all)
deploy(all)
